src/snowflake/snowpark/modin/plugin/extensions/index.py

```diff
@@ -501,12 +508,18 @@ def astype(self, dtype: Any, copy: bool = True) -> Index:
         >>> idx.astype('float')
         Index([1.0, 2.0, 3.0], dtype='float64')
         """
-        WarningMessage.index_to_pandas_warning("astype")
-        return Index(
-            self.to_pandas().astype(dtype=dtype, copy=copy),
-            dtype=dtype,
-            convert_to_lazy=self.is_lazy,
-        )
+        if dtype is not None:
```

Comment:
Discussed offline: We might need some tests to ensure this also works for non-lazy index. Otherwise Looks good!